app/api/scorecard.py

```diff
@@ -0,0 +1,140 @@
+from app import dependencies
+from fastapi import APIRouter, Depends
+from fastapi.encoders import jsonable_encoder
+from app import schemas, crud
+from typing import Any, List
+from app.utils.base import format_code, check_metric, stringify_value
+from .exceptions import HTTPResponseCustomized
+from .responses import ResponseCustomized
+from fastapi.responses import ORJSONResponse
+from app.schemas.apiResponse import CustomResponse
+from app.schemas.scorecardServiceMetric import ScorecardServiceMetricCreate
+from app.crud.scorecardServiceMetric import CRUDScoreCardServiceMetric
+
+
+router = APIRouter()
+@router.post("/", response_model=CustomResponse, response_class=ResponseCustomized)
+def createScoreCard(scoreCardInput: ScorecardServiceMetricCreate,
+                    scoreCardCrud: crud.CRUDScoreCard = Depends(
+                        dependencies.getScoreCardsCrud),
+                    serviceScorecardCrud: crud.CRUDMicroserviceScoreCard = Depends(
+                        dependencies.getMicroserviceScoreCardsCrud),
+                    scorecardMetricCrud: crud.CRUDScoreCardMetric = Depends(
+                        dependencies.getScoreCardMetricsCrud),
+                    serviceCrud: crud.CRUDMicroservice = Depends(
+                        dependencies.getMicroservicesCrud),
+                    metricCrud: crud.CRUDMetric = Depends(
+                        dependencies.getMetricsCrud),
+                    scorecard: crud.CRUDScoreCard = Depends(
+                        dependencies.getScoreCardsCrud)
+                    ):
+    # Create the scorecard.
+    scoreCardInput.code = format_code(scoreCardInput.name)
+    scorecard = schemas.ScoreCardCreate(
+        name=scoreCardInput.name,
+        code=scoreCardInput.code,
+        description=scoreCardInput.description
+    )
+    if (scoreCardCrud.getByScoreCardCode(scoreCardInput.code)):
+        raise HTTPResponseCustomized(
+            status_code=400, detail="Scorecard already found")
+    try:
+        obj = scoreCardCrud.create(scorecard)
+        scorecardID = obj.id
+    except Exception as e:
+        raise HTTPResponseCustomized(status_code=422, detail=f"Error: {e}")
+
+    # if the scorecardids are passed duplicated
+    servicesIDsSet = list(set(scoreCardInput.services))
+    # i need to check if the serviceid is found or not
+    service = serviceCrud.getByServiceIds(servicesIDsSet)
+    serviceListToBeCreated = []
+    for serviceid in servicesIDsSet:
+        if (len(service) == len(servicesIDsSet)):
+            serviceScorecard = schemas.MicroserviceScoreCardCreate(
+                scoreCardId=scorecardID,
+                microserviceId=serviceid
+            )
+            serviceListToBeCreated.append(serviceScorecard)
+        else:
+            scoreCardCrud.delete(scorecardID)
```

Comment:
this will be useless after you implement the changes requested in the previous two comments